### PR TITLE
chore: simplify new connect sdk flow

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/SA-499/simplify-new-connect-sdk-dialog-flow

This simplifies a few things about the new connect SDK flow.

Also includes some minor visual fixes.

### CODE REVIEW

File: frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.test.tsx [hunk 1]
```diff
@@ -1,62 +1,59 @@
...
+    renderComponent();
+
+    await waitFor(() => {
+        expect(
+            screen.getByRole('button', { name: /generate api key/i }),
+        ).toBeInTheDocument();
+    });
+});
+
+test('calls onApiKeyGenerated with token secret when clicking Next', async () => {
```
Comment (krzychukula) [inline frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.test.tsx]: I tested it and it can be quite confusing for the user that they have to click "Next" to have a different api key. 
1. pick sdk
2. select development environment
3. generate api key
4. click next
5. Go Back to the Generate API step
6. Change environment to production
7. Generate the new API key

Now you're looking at a new key, but if you don't click on "Next" then it's not used in practice. 
You can just lick on the "Configure the SDK" accordion at this point as it's not blocked in any way.

File: frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.tsx [hunk 2]
```diff
@@ -186,16 +108,75 @@ export const GenerateApiKey = ({
         }
     };
 
+    const longestEnv =
```
Comment (krzychukula) [inline frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.tsx:120]: IMHO. hiding this in a small component was better. Now we're looking at API and logic and it's mixed with how a select renders. Still, it's not a big deal if you strongly prefer it this way.

**Review by krzychukula — COMMENTED** [review]
I don't mind overall. But, there are two places I'm not sure about. See comments.

File: frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.test.tsx (hunk 1 above)
Comment (nunogois) [reply frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.test.tsx]: I'm not sure I agree. I don't think this very specific edge case is worth the extra complexity. But here: https://github.com/Unleash/unleash/pull/12024/commits/3b694d4f74aea67823bbe2d480a6799802016a10

File: frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.tsx (hunk 2 above)
Comment (nunogois) [reply frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.tsx:120]: I disagree, there was nothing unique about this select to warrant its own component. If we want the adaptable width behavior, then we should either:
- Adapt our current select components to support this behavior
- Create a new generic select component that supports this behavior

Given that adding a new component now means paying the cost of 3 files (component, test, story), even for extremely simple components, this feels a lot simpler and straight to the point.

Comment (krzychukula) [reply frontend/src/component/onboarding/dialog/NewConnectSdkDialog/GenerateApiKey/GenerateApiKey.tsx:120]: > Given that adding a new component now means paying the cost of 3 files (component, test, story), even for extremely simple components, this feels a lot simpler and straight to the point.

IMHO. 
If it's used in one place and it's covered by parent components tests then there is no need for a separate unit test. 
But, it makes sense to have dedicated tests for components that are used in multiple places to make those wrapping components tests simpler (they can depend on the components being tested). 

Separate story can help with reusability - it's going to be easier to notice we have 10 inline selects doing the same thing if they show up on the same list. 
But, I'm ok without it as well. 

IMHO. it still makes sense in a separate file. If you feel like you don't want to have 3 files for it, then just leave 1 :)

File: frontend/src/component/onboarding/dialog/NewConnectSdkDialog/NewConnectSdkDialog.tsx [hunk 3]
```diff
@@ -206,20 +205,25 @@ const InnerDialog = ({
...
+                        return (
+                            <ConnectSdkDialogStep
+                                key={title}
+                                stepNumber={index + 1}
+                                title={title}
+                                isExpanded={expandedStep === index}
+                                isCompleted={isCompleted}
+                                isDisabled={isDisabled}
+                                onExpand={() => setExpandedStep(index)}
+                                summary={isCompleted && summary}
```
Comment (nunogois) [inline frontend/src/component/onboarding/dialog/NewConnectSdkDialog/NewConnectSdkDialog.tsx:221]: Only mounts the summary if the step is completed, matching our designs in a more accurate manner.

File: frontend/src/component/onboarding/dialog/NewConnectSdkDialog/NewConnectSdkDialog.tsx [hunk 4]
```diff
@@ -171,31 +172,29 @@ const InnerDialog = ({
...
             title: 'Configure the SDK',
-            content: apiKey ? (
+            content: (
                 <ConfigureSdk
                     projectId={projectId}
                     sdk={sdk}
                     apiKey={apiKey}
                     feature={feature}
-                    expanded={expandedStep === 2}
+                    isActive={currentStep >= 2}
```
Comment (nunogois) [inline frontend/src/component/onboarding/dialog/NewConnectSdkDialog/NewConnectSdkDialog.tsx:194]: I know `expanded` was confusing, so I renamed this to `isActive`, which essentially means the step has started.

**Review by krzychukula — APPROVED** [review]
